app/javascript/components/map/MapPopup.vue

```diff
@@ -110,6 +100,16 @@ export default {
       this.randomKey += 1;
       this.data.datasets = [];
 
+      const ids =[]
+      this.chartData.forEach(item => ids.push(item.id))
+
+      for (const [key, value] of Object.entries(CHART_COLOURS)) {
+        // if using ids key has to be changed to Number type --> Number(key) https://stackoverflow.com/a/49550265/10897687
+        if (ids.includes(Number(key))) {
+          this.colors.push(value)
+        }
```

Comment:
do we want an else condition here that puts in a default colour of the ID isn't one in the colour hash?